src/basic/path-util.c

```diff
@@ -585,9 +585,9 @@ char* path_join_internal(const char *first, ...) {
         return joined;
 }
 
-int find_executable(const char *name, char **ret) {
+int find_executable_full(const char *name, bool use_path_envvar, char **ret) {
```

Comment:
maybe just take a string as second arg, and let the caller invoke getenv() or secure_getenv() or whatever it needs... shouldn't make things much longer to call this as `find_executable(…, getenv(…), …)`

Reply:
I agree this would be more elegant, but it would mean that we'd always call getenv(). Right now we skip that if the path is absolute. getenv() is potentially slow because it needs to scan the array. So I think it's better to leave it with the bool param.